plugins/tactics/src/Ide/Plugin/Tactic/CodeGen.hs

```diff
@@ -120,13 +121,13 @@ unzipTrace l =
 
 
 -- | Essentially same as 'dataConInstOrigArgTys' in GHC,
---   but we need some tweaks in GHC >= 8.8.
---   Since old 'dataConInstArgTys' seems working with >= 8.8,
---   we just filter out class dictionaries and coercions from the result.
+--  but only accepts universally quantified types as the second arguments
+--  and automatically introduces existentials.
 dataConInstOrigArgTys' :: DataCon -> [Type] -> [Type]
-dataConInstOrigArgTys' con ty =
-    let tys0 = dataConInstOrigArgTys con ty
-    in filter (not . isPredTy) tys0
+dataConInstOrigArgTys' con uniTys =
+  let exvars = dataConExTys con
+   in dataConInstOrigArgTys con $
+        uniTys ++ map mkTyVarTy exvars
```

Comment:
I think it matters. I couldn't find a precise specification for the input of `dataConInstOrigArgTys`, but [its implementation](https://hackage.haskell.org/package/ghc-8.10.1/docs/src/DataCon.html#local-6989586621680782681) places DataCon's existentials after universals and unifies them with the second argument (`uniTys ++ map mkTyVarTy exvars` here).

Reply:
Sounds good. Mind adding a quick comment about why this works and under what assumptions? I suspect GHC will probably break this upstream within a version or two, and it would be nice to have some idea why when that occurs :) Happy to merge afterwards!

Reply:
OK, I've just elaborated on it. Thanks for suggestions!